includes/functions.php

```diff
@@ -76,3 +76,136 @@ function get_stories_theme_support() {
 
 	return $theme_support;
 }
+
+/**
+ * Wrapper function for fetching field states
+ * based on the view types.
+ *
+ * Mainly uses FieldState and Fields classes.
+ *
+ * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+ *
+ * @return array
+ */
+function fields_states() {
+	$views = get_layouts();
+
+	$fields = [
+		'title',
+		'author',
+		'date',
+		'image_align',
+		'excerpt',
+		'archive_link',
+	];
+
+	$field_states = [];
+
+	foreach ( $views as $view_type => $view_label ) {
+		$field_state = ( new Story_Query( [ 'view_type' => $view_type ] ) )->get_renderer()->field();
+		foreach ( $fields as $field ) {
+			$field_states[ $view_type ][ $field ] = [
+				'show'     => $field_state->$field()->show(),
+				'label'    => $field_state->$field()->label(),
+				'readonly' => $field_state->$field()->readonly(),
+			];
+		}
+	}
+
+	return $field_states;
+}
+
+/**
+ * Put some tinymce related data on the page.
+ *
+ * @param string $hook Hook name for current page.
+ *
+ * @return void
+ */
+function web_stories_script_data( $hook ) {
+	if ( 'widgets.php' !== $hook ) {
+		return;
+	}
+
+	$order      = get_orderby();
+	$views      = get_layouts();
+	$order_list = [];
+	$view_types = [];
+
+	foreach ( $order as $order_key => $an_order ) {
+		$order_list[] = [
+			'label' => $an_order,
+			'value' => $order_key,
+		];
+	}
+
+	foreach ( $views as $view_key => $view_label ) {
+		$view_types[] = [
+			'label' => $view_label,
+			'value' => $view_key,
+		];
+	}
+
+	$field_states = fields_states();
+
+	$data = [
+		'orderlist' => $order_list,
+		'icon'      => WEBSTORIES_PLUGIN_DIR_URL . '/__static__/carousel-icon.png',
+		'tag'       => 'stories',
+		'views'     => $view_types,
+		'fields'    => $field_states,
+	];
+
+	echo "<script type='text/javascript'>\n";
+	echo 'var webStoriesData = ' . wp_json_encode( $data ) . ';';
+	echo "\n</script>";
+}
+add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\web_stories_script_data' );
```

Comment:
I really think we should use `wp_localize_script` here.

Reply:
@spacedmonkey  
We could have used `wp_localize_script`, but this data is required on multiple pages:

1. Widget page.
2. Post edit page.
3. New post page.

This data does not have dependency on a single script, so instead we can use `admin_print_scripts-widgets.php` hook to add this data to page and later use it in different places.

Reply:
You should still use `wp_localize_script`. This should not live as a function as well. This should be a class with a method. Add wp_localize_script for scripts in all those contexts.